pkg/services/publicdashboards/metric/metric.go

```diff
@@ -0,0 +1,68 @@
+package metric
+
+import (
+	"context"
+	"errors"
+	"strconv"
+	"time"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/services/publicdashboards"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type PublicDashboardsMetricServiceImpl struct {
+	store   publicdashboards.Store
+	Metrics *Metrics
+	log     log.Logger
+}
+
+func ProvideService(
+	store publicdashboards.Store,
+) (*PublicDashboardsMetricServiceImpl, error) {
+	s := &PublicDashboardsMetricServiceImpl{
+		store:   store,
+		Metrics: NewMetrics(),
+		log:     log.New("publicdashboards.metric"),
+	}
+
+	if err := s.registerMetrics(); err != nil {
+		return nil, err
+	}
+
+	return s, nil
+}
+
+func (s *PublicDashboardsMetricServiceImpl) registerMetrics() error {
+	err := prometheus.Register(s.Metrics.PublicDashboardsTotal)
+	var alreadyRegisterErr prometheus.AlreadyRegisteredError
+	if errors.As(err, &alreadyRegisterErr) {
+		if alreadyRegisterErr.ExistingCollector == alreadyRegisterErr.NewCollector {
+			err = nil
+		}
+	}
+
+	return err
+}
+
+func (s *PublicDashboardsMetricServiceImpl) Run(ctx context.Context) error {
+	ticker := time.NewTicker(12 * time.Hour)
```

Comment:
I believe this won't run until after the first 12 hours have passed, is that what you want?

Reply:
We talked to @thanos-karachalios and we agree to run this metric every 12 hours. 
maybe decrease to 6 hours?

Reply:
I don't think the frequency is a problem, but you don't initialize it until after the first 12 hours have passed as far as I can tell.

Reply:
got it, gonna run it once and then every 12 hours